src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -249,58 +366,63 @@ public virtual bool TryFindConnection(string ownerUri, out ConnectionInfo connec
                 {
                     // Only remove the token from the map if it is the same one created by this request
                     CancellationTokenSource sourceValue;
-                    if (ownerToCancellationTokenSourceMap.TryGetValue(connectionParams.OwnerUri, out sourceValue) && sourceValue == source)
+                    if (cancelTupleToCancellationTokenSourceMap.TryGetValue(cancelKey, out sourceValue) && sourceValue == source)
                     {
-                        ownerToCancellationTokenSourceMap.TryRemove(connectionParams.OwnerUri, out sourceValue);
+                        cancelTupleToCancellationTokenSourceMap.TryRemove(cancelKey, out sourceValue);
                     }
                 }
             }
 
-            ownerToConnectionMap[connectionParams.OwnerUri] = connectionInfo;
+            // Return null upon success
+            return null;
+        }
 
-            // Update with the actual database name in connectionInfo and result
-            // Doing this here as we know the connection is open - expect to do this only on connecting
-            connectionInfo.ConnectionDetails.DatabaseName = connectionInfo.SqlConnection.Database;
-            response.ConnectionSummary = new ConnectionSummary
+        /// <summary>
+        /// Gets the existing connection with the given URI and connection type string. If none exists, 
+        /// creates a new connection. This cannot be used to create a default connection or to create a 
+        /// connection if a default connection does not exist.
+        /// </summary>
+        public async Task<DbConnection> GetOrOpenConnection(string ownerUri, string connectionType)
+        {
+            if (String.IsNullOrEmpty(ownerUri) || String.IsNullOrEmpty(connectionType))
             {
-                ServerName = connectionInfo.ConnectionDetails.ServerName,
-                DatabaseName = connectionInfo.ConnectionDetails.DatabaseName,
-                UserName = connectionInfo.ConnectionDetails.UserName,
-            };
+                return null;
+            }
+            if (String.Equals(connectionType, ConnectionType.Default))
```

Comment:
Generally unless you need case insensitive or other special case comparisons, you can just do `str1 == str2`

Although this code would also be neater if we were using enums